ApplensBackend/Controllers/CaseCleansingController.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Authorization;
+using System.Data.SqlClient;
+using Dapper;
+using Dapper.Contrib.Extensions;
+using Kusto.Data.Net.Client;
+using System.Dynamic;
+using Kusto.Data;
+using Microsoft.Extensions.Configuration;
+
+namespace AppLensV3.Controllers
+{
+    [Produces("application/json")]
+    [Route("api/casecleansing")]
+    [Authorize]
+    public class CaseCleansingController : Controller
+    {
+        public class CaseSimple
+        {
+            public string IncidentId { get; set; }
+            public DateTime Time { get; set; }
+            public string Status { get; set; }
+            public string AssignedTo { get; set; }
+            public DateTime ClosedTime { get; set; }
+            public int ID { get; set; }
+            public int RecommendationCount { get; set; }
+            public string Title { get; set; }
+        }
+
+        public class CaseExtra
+        {
+            public Recommendation[] Recommendations { get; set; }
+            public dynamic KustoData { get; set; }
+        }
+
+        public class Recommendation
+        {
+            public string RuleName { get; set; }
+            public string OldClosedAgainst { get; set; }
+            public string RecommendedClosedAgainst { get; set; }
+            public DateTime RecommendationDate { get; set; }
+        }
+
+        private IConfiguration _configuration;
+        private class CaseCleansingConfiguration
+        {
+            public string ConnectionString { get; set; }
+            public string KustoP360ConnectionString { get; set; }
+        }
+
+        private readonly CaseCleansingConfiguration _caseCleansingConfiguration;
+
+        public CaseCleansingController(IConfiguration configuration)
+        {
+            _configuration = configuration;
+            _caseCleansingConfiguration = new CaseCleansingConfiguration();
+            configuration.GetSection("CaseCleansing").Bind(_caseCleansingConfiguration);
+        }
+
+        [HttpGet("getallcases")]
+        public CaseSimple[] Get()
+        {
+            string str = @"select IncidentId, Time, Status, AssignedTo, ClosedTime, Incidents.ID As ID, Count(RuleID) As RecommendationCount, Title
```

Comment:
nitpick: Maybe move these queries to some constants/queries class, just since the justification looks weird here.